core/src/main/java/org/elasticsearch/common/xcontent/XContentType.java

```diff
@@ -133,7 +159,22 @@ public static XContentType fromMediaTypeOrFormat(String mediaType) {
         return null;
     }
 
-    private static boolean isSameMediaTypeAs(String stringType, XContentType type) {
+    /**
+     * Attempts to match the given media type with the known {@link XContentType} values. This match is done in a case-insensitive manner.
+     * The provided media type should not included any parameters. This method is suitable for parsing part of the {@code Content-Type}
```

Comment:
I will take strict out of the name. Strict refers to how the value is compared and that it does not accept things like "json" but since the other method was renamed